wgpu-core/src/lib.rs

```diff
@@ -163,17 +179,22 @@ impl Drop for MultiRefCount {
 }
 
 #[derive(Debug)]
-struct LifeGuard {
+pub struct LifeGuard {
     ref_count: Option<RefCount>,
     submission_index: AtomicUsize,
+    #[cfg(debug_assertions)]
+    pub(crate) label: String,
 }
 
 impl LifeGuard {
-    fn new() -> Self {
+    #[allow(unused_variables)]
+    fn new(label: String) -> Self {
```

Comment:
Can we instead do something like this:
```rust
pub fn new(label: &str) -> Self {
// something under `cfg(debug_assertions)` inside
}
```

This would allow all the call sites to not care about `debug_assertions`

Reply:
I don't understand, how does the callsite here care about `debug_assertions`?

It could be `&str` though I guess

Reply:
Ah, I misread. It will only affect the call sites with static strings, like
```rust
            #[cfg(debug_assertions)]
            life_guard: LifeGuard::new("<device>".to_string()),
            #[cfg(not(debug_assertions))]
            life_guard: LifeGuard::new(String::new()),
```